host-controller/src/main/java/org/jboss/as/host/controller/operations/StaticDiscoveryWriteAttributeHandler.java

```diff
@@ -0,0 +1,96 @@
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.host.controller.operations;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.WRITE_ATTRIBUTE_OPERATION;
+
+import java.util.function.Predicate;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+
+/**
+ * This is an alias for elements in the options element, to perform a write attribute operation we need to update the parent
+ * discovery-options/static-discovery=primary:write-attribute(name=port,value=9999)
+ * should be treated like the following
+ * /host=slave/core-service=discovery-options:write-attribute(name=options,value=[{static-discovery={name=primary,
+ * protocol="${jboss.domain.master.protocol:remote+http}",host="${jboss.domain.master.address}",port=9999}}]
+ *
+ * @author tmiyar
+ *
+ */
+public class StaticDiscoveryWriteAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+    public StaticDiscoveryWriteAttributeHandler(SimpleAttributeDefinition attribute) {
+        super(attribute);
+    }
+
+    @Override
+    protected boolean requiresRuntime(OperationContext context) {
+        // HCs may connect to the in either RunningMode.NORMAL or ADMIN_ONLY,
+        // so the running mode doesn't figure in whether reload is required
+        return !context.isBooting();
+    }
+
+    @Override
+    protected void finishModelStage(OperationContext context, ModelNode operation, String attributeName, ModelNode newValue,
+            ModelNode oldValue, Resource model) throws OperationFailedException {
+        super.finishModelStage(context, operation, attributeName, newValue, oldValue, model);
+
+        final PathAddress operationAddress = PathAddress.pathAddress(operation.get(OP_ADDR));
+        final PathAddress discoveryOptionsAddress = operationAddress.subAddress(0, operationAddress.size() - 1);
+        final ModelNode discoveryOptions = Resource.Tools.readModel(context.readResourceFromRoot(discoveryOptionsAddress));
+
+        // Get the current list of discovery options and set new value for attribute
+        final ModelNode currentList = discoveryOptions.get(ModelDescriptionConstants.OPTIONS);
```

Comment:
If you get rid of the Resource.Tools.readModel per my comment above you should clone this though; otherwise you are manipulating the underlying model. Resource.Tools.readModel clones the whole thing internally.

final ModelNode currentList = discoveryOptions.get(ModelDescriptionConstants.OPTIONS).clone();